src/ReactiveDomain.Users/Messages/UserMsgs.cs

```diff
@@ -0,0 +1,341 @@
+﻿using System;
+using ReactiveDomain.Messaging;
+
+namespace ReactiveDomain.Users.Messages
+{
+
+    /// <summary>
+    /// Messages for the User domain.
+    /// </summary>
+    public class UserMsgs
+    {
+        /// <summary>
+        /// Create a new user.
+        /// </summary>
+        public class CreateUser : Command
+        {
+            /// <summary>The unique ID of the new user.</summary>
+            public readonly Guid UserId;
+            /// <summary>The user's given name. This is the first name in most cultures.</summary>
+            public readonly string GivenName;
+            /// <summary>The user's surname or family name. This is the last name in most cultures.</summary>
+            public readonly string Surname;
+            /// <summary>The user's email address.</summary>
+            public readonly string Email;
+            /// <summary> User full name for local windows users</summary>
+            public readonly string FullName;
+
+            /// <summary>
+            /// Create a new user.
+            /// </summary>
+            /// <param name="userId">The unique ID of the new user.</param>         
+            /// <param name="givenName">The user's given name. This is the first name in most cultures.</param>
+            /// <param name="surname">The user's surname or family name. This is the last name in most cultures.</param>
+            /// <param name="email">The user's email address.</param>
+            /// <param name="fullName">user full name.</param>
+            public CreateUser(
+                Guid userId,
+                string fullName,
+                string givenName,
+                string surname,
+                string email
+                )
+            {
+                UserId = userId;
+                FullName = fullName;
+                GivenName = givenName;
+                Surname = surname;
+                Email = email;
+
+            }
+        }
+        public class UserEvent : Event
```

Comment:
If we're going to bother defining this type, would it make sense to set the `UserId` in this base class's ctor?